src/update-manager.ts

```diff
@@ -571,27 +606,29 @@ export default class UpdateManager {
       return false
     }
 
-    wssURI = uriJoin(wssURI, doc.uri)
-    wssURI = wssURI.replace(/^http:/, 'ws:').replace(/^https:/, 'wss:')
+    wssURI = uriJoin(wssURI, doc.value)
+    const validWssURI = wssURI.replace(/^http:/, 'ws:').replace(/^https:/, 'wss:')
     console.log('Web socket URI ' + wssURI)
 
     var openWebsocket = function () {
       // From https://github.com/solid/solid-spec#live-updates
       var socket
       if (typeof WebSocket !== 'undefined') {
-        socket = new WebSocket(wssURI)
+        socket = new WebSocket(validWssURI)
+        //@ts-ignore Firefox Addon
       } else if (typeof Services !== 'undefined') { // Firefox add on http://stackoverflow.com/questions/24244886/is-websocket-supported-in-firefox-for-android-addons
+        //@ts-ignore Firefox Addon
         socket = (Services.wm.getMostRecentWindow('navigator:browser').WebSocket)(wssURI)
```

Comment:
Given that Firefox Add-Ons now use WebExtensions, I presume that this entire clause can be deleted :)